File: backend/MiniBilge.Application/DTOs/Profile/CreateChildProfileRequest.cs
namespace MiniBilge.Application.DTOs.Profile;

public class CreateChildProfileRequest
{
    public string Name { get; set; } = string.Empty;
    public DateTime DateOfBirth { get; set; }
    public int GradeLevel { get; set; }
    public int? EnglishLevel { get; set; }
    public string? AvatarImageUrl { get; set; }
    /// <summary>0 = Offline, 1 = Online. Belirtilmezse 0 (Offline) kullanılır.</summary>
    public int PodcastListeningMode { get; set; } = 0;
    public bool IsTeacher { get; set; } = false;
}

File: mobile/minibilge_mobile/lib/features/child_profile/models/create_child_profile_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'create_child_profile_request.freezed.dart';
part 'create_child_profile_request.g.dart';

@freezed
class CreateChildProfileRequest with _$CreateChildProfileRequest {
  const factory CreateChildProfileRequest({
    @JsonKey(name: 'Name') required String name,
    @JsonKey(name: 'DateOfBirth') required DateTime dateOfBirth,
    @JsonKey(name: 'GradeLevel') required int gradeLevel,
    @JsonKey(name: 'EnglishLevel') int? englishLevel,
    @JsonKey(name: 'AvatarImageUrl') String? avatarImageUrl,
    @JsonKey(name: 'PodcastListeningMode') @Default(0) int podcastListeningMode,
  }) = _CreateChildProfileRequest;

  factory CreateChildProfileRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateChildProfileRequestFromJson(json);
}
